pyomo/contrib/alternative_solutions/solnpool.py

```diff
@@ -0,0 +1,115 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2024
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from pyomo.common.dependencies import attempt_import
+
+gurobipy, gurobipy_available = attempt_import("gurobipy")
+
+import pyomo.environ as pe
+from pyomo.contrib import appsi
+import pyomo.contrib.alternative_solutions.aos_utils as aos_utils
+from pyomo.contrib.alternative_solutions import Solution
+
+
+def gurobi_generate_solutions(
+    model,
+    *,
+    num_solutions=10,
+    rel_opt_gap=None,
+    abs_opt_gap=None,
+    solver_options={},
+    tee=False,
+    quiet=True,
+):
+    """
+    Finds alternative optimal solutions for discrete variables using Gurobi's
+    built-in Solution Pool capability. See the Gurobi Solution Pool
+    documentation for additional details.
+
+    Parameters
+    ----------
+    model : ConcreteModel
+        A concrete Pyomo model.
+    num_solutions : int
+        The maximum number of solutions to generate. This parameter maps to
+        the PoolSolutions parameter in Gurobi.
+    rel_opt_gap : non-negative float or None
+        The relative optimality gap for allowable alternative solutions.
+        None implies that there is no limit on the relative optimality gap
+        (i.e. that any feasible solution can be considered by Gurobi).
+        This parameter maps to the PoolGap parameter in Gurobi.
+    abs_opt_gap : non-negative float or None
+        The absolute optimality gap for allowable alternative solutions.
+        None implies that there is no limit on the absolute optimality gap
+        (i.e. that any feasible solution can be considered by Gurobi).
+        This parameter maps to the PoolGapAbs parameter in Gurobi.
+    solver_options : dict
+        Solver option-value pairs to be passed to the Gurobi solver.
+    tee : boolean
+        Boolean indicating that the solver output should be displayed.
+    quiet : boolean
+        Boolean indicating whether to suppress all output.
+
+    Returns
+    -------
+    solutions
+        A list of Solution objects.  [Solution]
+    """
+    #
+    # Setup gurobi
+    #
+    if not gurobipy_available:
+        return []
+    opt = appsi.solvers.Gurobi()
+    if not opt.available():  # pragma: no cover
+        return []
```

Comment:
And same here? This should probably throw an error saying that appsi needs to be installed, yes?

Reply:
See above.  We're being consistent in the current API.